db/rocksdb_ethereumtype.go

```diff
@@ -347,9 +431,15 @@ func (d *RocksDB) addToAddressesAndContractsEthereumType(addrDesc bchain.Address
 		if ac == nil {
 			ac = &AddrContracts{}
 		}
-		addressContracts[strAddrDesc] = ac
+		if addressContracts != nil {
```

Comment:
Is this condition necessary? Why would we pass nil `addressContracts`? I do not see any change in the calling of `addToAddressesAndContractsEthereumType`

Reply:
The reason for this is due to `addressContracts` being moved into the `RocksDB` struct and used for both `bulkconnect` and `rocksdb_ethereumtype`. This condition is to handle `nil` being passed from `connectBlockEthereumType` as we don't need a second map being assigned for bulk connect. The local `addrContracts` variable is solely to track `addrContracts` for a single block in the serial sync flow.

Reply:
In other words, the `RocksDB.addressContracts` is the cache state variable used by `bulkconnect` always and only, and can be used by the serial sync if configured via `-dbmaxaddrcontracts`. The `addressContracts` param is used only for serial sync to track and store the associated address contracts for each block.

Reply:
OK, got it. Thank you.